### PR TITLE
book: fix mistake (File::open -> File::create)

### PR DESCRIPTION
**Pull request by danclive**

### CODE REVIEW

**Comment by steveklabnik** [comment]
Thank you! Why is this a mistake, though? And if it is, then the error message needs to change as well.

**Comment by dangcheng** [comment]
In this example, we are going to open the file and write it in the file. Open is read-only, so it needs to be changed to create to write.

File: src/doc/book/traits.md [hunk 1]
```diff
@@ -275,7 +275,7 @@ won’t have its methods:
 [write]: ../std/io/trait.Write.html
 
 ```rust,ignore
-let mut f = std::fs::File::open("foo.txt").expect("Couldn’t open foo.txt");
+let mut f = std::fs::File::create("foo.txt").expect("Couldn’t open foo.txt");
```
Comment (frewsxcv) [inline src/doc/book/traits.md]: I think @steveklabnik is suggesting `"Couldn’t open foo.txt"` should be changed to `"Couldn’t create foo.txt"`

**Comment by dangcheng** [comment]
Thank you! @frewsxcv

**Comment by steveklabnik** [comment]
@bors: r+ rollup

ah ha! Since it is marked `ignore` anyways, I didn't notice. Thank you so much!

**Pull request merged**